src/navigation.ts

```diff
@@ -233,33 +198,45 @@ export class Navigator extends GObject.Object {
 
     this.last_controller = new AbortController();
 
-    let response: ReturnType<Endpoint<Gtk.Widget>["load"]>;
+    const response = endpoint.load(component, {
+      match: match as MatchResult<Record<string, string>>,
+      url,
+      signal: this.last_controller.signal,
+    });
 
-    try {
-      response = endpoint.load(component, {
-        match: match as MatchResult<Record<string, string>>,
-        url,
-        signal: this.last_controller.signal,
-      });
+    if (!response) return null;
 
-      if (!response) return null;
+    this.loading = true;
 
-      // temporarily show an old page if it's available
-      if (this._stack.get_child_by_name(uri)) {
-        this._stack.set_visible_child_name(uri);
-      }
+    const page = new Page();
+    page.title = endpoint.title;
 
-      this.loading = true;
+    if (url.searchParams.has("replace")) {
+      this._view.animate_transitions = false;
+      this._view.pop();
+      this._view.push(page);
+    } else {
+      this._view.animate_transitions = true;
+      this._view.push(page);
+    }
```

Comment:
Can simplify; both branches are pushing the page onto the view

```suggestion
    if (url.searchParams.has("replace")) {
      this._view.animate_transitions = false;
      this._view.pop();
    } else {
      this._view.animate_transitions = true;
    }
    this._view.push(page);
```